File: spring-web-java/src/main/java/spring/web/java/global/config/security/UserDetailsServiceImpl.java
package spring.web.java.global.config.security;

import org.springframework.http.HttpStatus;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;

import lombok.RequiredArgsConstructor;
import spring.web.java.domain.member.repository.MemberRepository;
import spring.web.java.global.common.ResponseMessage;
import spring.web.java.global.exception.DomainException;

@RequiredArgsConstructor
public class UserDetailsServiceImpl implements UserDetailsService {

	private final MemberRepository memberRepository;

	@Override
	public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
		return memberRepository.findByAccount(username)
			.map(UserDetailsImpl::new)
			.orElseThrow(() -> new DomainException(ResponseMessage.DATA_NOT_FOUND, HttpStatus.NOT_FOUND));
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/global/config/security/UserDetailsServiceImpl.kt
package spring.web.kotlin.global.config.security

import org.springframework.http.HttpStatus
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import spring.web.kotlin.domain.member.repository.MemberRepository
import spring.web.kotlin.global.common.ResponseMessage
import spring.web.kotlin.global.exception.DomainException

class UserDetailsServiceImpl(
    private val memberRepository: MemberRepository
) : UserDetailsService {
    override fun loadUserByUsername(username: String): UserDetails {
        val member = (memberRepository.findByAccount(username)
            ?: throw DomainException(ResponseMessage.DATA_NOT_FOUND, HttpStatus.NOT_FOUND))

        return UserDetailsImpl(member)
    }
}